### PR TITLE
Added support for log management calls.

### PR DESCRIPTION
**Pull request by ellynch-r7**
- get logs
 - create logs
 - delete logs
 - rename logs
 - update logs
 - replace logs

**NOTE**: The update & rename calls require [LOG-8550](https://logentries.atlassian.net/browse/LOG-8850) to work properly. At the moment, these calls are returning 405 for valid requests.

### CODE REVIEW

File: lecli/cli.py
Comment (udangel-r7) [inline lecli/cli.py]: how about `click.echo("Example usage: lecli createlog -n 'new_log_name'")` however, why do we need to have single quotes around `new_log_name`

File: lecli/cli.py
Comment (udangel-r7) [inline lecli/cli.py]: We should use `sys.exit` vs. `exit` (seems like this is wrong in whole lecli)

File: README.md
Comment (stopal-r7) [inline README.md]: typo: path

File: lecli/cli.py
Comment (stopal-r7) [inline lecli/cli.py]: I assume this is "full path" of the file containing json, or can it be also relative? I think this information should be included in the documentation and help text.

File: lecli/cli.py
Comment (stopal-r7) [inline lecli/cli.py]: here you can use double quote and single quote "Example usage: lecli createlog -n 'new_log_name'", that way you won't need to use escape character

File: lecli/cli.py
Comment (udangel-r7) [inline lecli/cli.py]: maybe provide a better example and it should probably be printed to sys.stderr instead of stdout

File: lecli/log_api.py
Comment (udangel-r7) [inline lecli/log_api.py]: can we make that configurable? because we will soon have multiple rest endpoints?

File: lecli/log_api.py
Comment (udangel-r7) [inline lecli/log_api.py]: how about changing it to something like:

```
key_name = 'log'
key_name = 'logs' if 'logs' in response.json()
print json.dumps(response.json()[key_name], indent=4, sort_keys=True)
```

File: lecli/log_api.py
Comment (udangel-r7) [inline lecli/log_api.py]: why the call to str? Won't this fail with unicode characters?

File: lecli/log_api.py
Comment (udangel-r7) [inline lecli/log_api.py]: indentation for `str(log_id)` also it would be probably cleaner to use something like `os.path.join(_url(), log_id)`

File: lecli/cli.py
Comment (ellynch-r7) [reply lecli/cli.py]: Yeah, they're not actually required at all. I just added them out of habit.

I'll update this and check the docs for it too.

File: lecli/api_utils.py
Comment (udangel-r7) [inline lecli/api_utils.py]: i think this is wrong as it will have 3 newlines instead of 2?

File: lecli/log_api.py
Comment (udangel-r7) [inline lecli/log_api.py]: as this is in `log_api`, shouldn't things like exit, printing etc. be handled by the caller and this just does the calls / returns the data?

File: lecli/log_api.py
Comment (stopal-r7) [reply lecli/log_api.py]: how about this?
```
key_name = 'logs' if 'logs' in response.json() else 'log'
print json.dumps(response.json()[key_name], indent=4, sort_keys=True)
```

i think existing code is much more readable though.

File: lecli/log_api.py
Comment (stopal-r7) [reply lecli/log_api.py]: @udangel-r7 I agree, i think we may even have a bug around encoding strings like this. look at: https://github.com/rapid7/lecli/issues/31

File: lecli/log_api.py
Comment (stopal-r7) [reply lecli/log_api.py]: i think os.path.join is for OS file system paths, for web URLs something like `"/".join([_url(), log_id])` sounds like the right call.

File: lecli/cli.py
Comment (ellynch-r7) [reply lecli/cli.py]: I've updated this (and error printing) throughout the project now

File: lecli/log_api.py
Comment (ellynch-r7) [reply lecli/log_api.py]: Should we do this for all of the endpoints?

At the moment, they're all hard-coded.

File: lecli/log_api.py
Comment (ellynch-r7) [reply lecli/log_api.py]: I don't think there's any harm in this being a tad more concise, I'll update it.

File: lecli/log_api.py
Comment (ellynch-r7) [reply lecli/log_api.py]: fixed

File: lecli/log_api.py
Comment (ellynch-r7) [reply lecli/log_api.py]: At the moment, this is very much in the style of the rest of the project. I think if we want to change it, it makes sense to change it elsewhere in the project too.

I don't mind doing this work, but given that we're OOO tomorrow, I think this might make the card bleed into next sprint a bit.

Thoughts?

File: lecli/cli.py
Comment (ellynch-r7) [reply lecli/cli.py]: It can be relative too. I'll update docs & help text

File: lecli/log_api.py
Comment (stopal-r7) [reply lecli/log_api.py]: Makes much more sense and gives more flexibility to separate view and data as @udangel-r7 suggests; however, I agree on timing with @ellynch-r7.

File: tests/test_log_api.py
Comment (stopal-r7) [inline tests/test_log_api.py]: I think here you can reuse the response body in https://github.com/rapid7/lecli/pull/32/files#diff-68084ff0bea49386c8ff8d6bd61ce0e9R143

File: tests/test_log_api.py
Comment (stopal-r7) [inline tests/test_log_api.py:72]: same here

File: lecli/log_api.py
Comment (udangel-r7) [reply lecli/log_api.py]: @stopal-r7 your is a little bit dense and i haven't seen that pattern before that much in the python world.

**Comment by ellynch-r7** [comment]
I've addressed the majority of the comments here in the latest push.

The only outstanding issue is the separation of concerns when it comes to error handling etc within the api modules. I've spoken to @dtracey-r7  about this, and we've agreed to push this out into it's own card.

**Review by stopal-r7 — APPROVED** [review]
LGTM when travis is happy.

File: lecli/log_api.py
Comment (scawley-r7) [inline lecli/log_api.py]: no need to call str() here

File: README.md
Comment (scawley-r7) [inline README.md:64]: I'm not sure this will display as intended. not 100% though..

File: lecli/_version.py
Comment (scawley-r7) [inline lecli/_version.py:4]: bumping by 2?

File: lecli/cli.py
Comment (scawley-r7) [reply lecli/cli.py]: How does this handle file not found?

File: README.md
Comment (ellynch-r7) [reply README.md:64]: It seems ok in the pycharm preview

File: lecli/_version.py
Comment (ellynch-r7) [reply lecli/_version.py:4]: There's another PR to go in before this one (#33) which is v 0.6.2

File: lecli/cli.py
Comment (ellynch-r7) [reply lecli/cli.py]: It doesn't, I'll update it now

**Review by scawley-r7 — APPROVED** [review]
Once travis is a happy camper

**Comment by ellynch-r7** [comment]
@dtracey-r7 @udangel-r7 

The update & rename methods had to be reworked. There's currently no support for PATCH requests in lesever/lerest. 

I've re-written those methods to use PUT requests. LECLI now does a bit of grunt work to merge the existing resource with the user-specified changes to make the calls behave as a PATCH would.

**Comment by ellynch-r7** [comment]
Putting this PR on hold pending some refactoring

**Pull request closed without merging**
